### PR TITLE
feat(core): add oidc config to the response of application apis

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(core): add oidc config to the response of application apis

## Linear Issue Reference

* LOG-1908

## Testing

UT & Tested in the admin console.

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 1]
```diff
@@ -170,13 +163,13 @@ const ApplicationDetails = () => {
     </>
   );
 
-  const AdvancedSettingsPage = oidcConfig && (
+  const AdvancedSettingsPage = data && (
     <>
       <FormField title="admin_console.application_details.token_endpoint">
-        <CopyToClipboard className={styles.textField} value={oidcConfig.token_endpoint} />
+        <CopyToClipboard className={styles.textField} value={data.oidcConfig.tokenEndpoint} />
```
Comment (simeng-li) [inline packages/console/src/pages/ApplicationDetails/index.tsx:169]: should we change the default oidcConfig key case behavior?

Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:169]: per offline discussion, keep camel case in our logto-core APIs.

File: packages/core/src/routes/application.ts [hunk 2]
```diff
@@ -27,9 +32,14 @@ export default function applicationRoutes<T extends AuthedRouter>(router: T) {
       findAllApplications(limit, offset),
     ]);
 
+    const oidcConfig = await got(discoveryUrl).json<SnakeCaseOidcConfig>();
```
Comment (wangsijie) [inline packages/core/src/routes/application.ts:35]: Can we fetch data from `oidc-provider` or some `node` native funtions?

Comment (IceHe) [inline packages/core/src/routes/application.ts:35]: better use promise.all
```ts
    const [{ count }, applications, oidcConfig] = await Promise.all([
      findTotalNumberOfApplications(),
      findAllApplications(limit, offset),
      got(discoveryUrl).json<SnakeCaseOidcConfig>(),
    ]);
```

File: packages/core/src/routes/application.ts [hunk 3]
```diff
@@ -45,14 +55,21 @@ export default function applicationRoutes<T extends AuthedRouter>(router: T) {
...
-      ctx.body = await insertApplication({
+      const application = await insertApplication({
         id: applicationId(),
         type,
         name,
         oidcClientMetadata: buildOidcClientMetadata(oidcClientMetadata),
         ...rest,
       });
 
+      const oidcConfig = await got(discoveryUrl).json<SnakeCaseOidcConfig>();
```
Comment (IceHe) [inline packages/core/src/routes/application.ts:66]: ditto

File: packages/core/src/routes/application.ts [hunk 4]
```diff
@@ -67,7 +84,14 @@ export default function applicationRoutes<T extends AuthedRouter>(router: T) {
         params: { id },
       } = ctx.guard;
 
-      ctx.body = await findApplicationById(id);
+      const application = await findApplicationById(id);
+
+      const oidcConfig = await got(discoveryUrl).json<SnakeCaseOidcConfig>();
```
Comment (IceHe) [inline packages/core/src/routes/application.ts:89]: ditto

File: packages/core/src/routes/application.ts [hunk 5]
```diff
@@ -85,10 +109,17 @@ export default function applicationRoutes<T extends AuthedRouter>(router: T) {
         body,
       } = ctx.guard;
 
-      ctx.body = await updateApplicationById(id, {
+      const application = await updateApplicationById(id, {
         ...body,
       });
 
+      const oidcConfig = await got(discoveryUrl).json<SnakeCaseOidcConfig>();
```
Comment (IceHe) [inline packages/core/src/routes/application.ts:116]: ditto

File: packages/core/src/routes/application.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/core/src/routes/application.ts:35]: No, I have researched.

**Review by IceHe — APPROVED** [review]
LGTM

**Review by wangsijie — APPROVED** [review]
Overall LGTM, better find another way to get oidc configs instead of http request to localhost.

**Pull request merged**